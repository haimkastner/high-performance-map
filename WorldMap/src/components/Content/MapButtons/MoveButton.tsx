import React from "react";
import { MapButtonIcon } from "./MapButtonIcon";
import { Box, Button } from "@material-ui/core";
import { setSharedState } from "../../../services/stats-store";
import { Platform } from "../../../models/models";

export class MoveButton extends React.Component {
  dir = 50;
  selectPlatform = () => {
    const platform: Platform = {
      ID: 5555,
      Name: "5555",
      PlatformType: "Aircraft",
      Spacial: {
        Movement: {
          Course: this.dir++,
          Speed: 400
        },
        Position: {
          Altitude: 0,
          Latitude: 70,
          Longitude: 44
        }
      }
    };
    setSharedState("selectedPlatform", platform);
    setSharedState("showNavBar", true);
    
  };

  render() {
    return (
      <Box>
        <Button onClick={this.selectPlatform}>
          <MapButtonIcon
            imageName={"sensors-move@2x.png"}
            width="19"
            height="19"
          ></MapButtonIcon>
        </Button>
      </Box>
    );
  }
}
