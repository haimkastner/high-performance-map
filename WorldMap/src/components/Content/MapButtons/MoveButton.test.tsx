import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MoveButton } from './MoveButton';

describe('MoveButton Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MoveButton />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<MoveButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // tslint:disable-next-line: no-empty
  describe('MoveButton Component Logic', () => {});
});
