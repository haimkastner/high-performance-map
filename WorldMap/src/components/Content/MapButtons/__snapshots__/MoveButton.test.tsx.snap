// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveButton Component matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-37"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="MuiBox-root MuiBox-root-70"
      >
        <img
          height="19"
          src="sensors-move@2x.png"
          width="19"
        />
      </div>
    </span>
  </button>
</div>
`;
